### ISSUE TITLE
Plugin fails on error with Volumio 2.175

### ISSUE THREAD

**Issue by Saiyato** [balbuze/volumio-plugins#53]
Not sure if I should raise the issue here, but maybe you have made some changes in the version of spotify-connect you deploy already.

There's a mixer error when the Boss DAC is selected:

```
Loading Spotify library...
Established under name 'VolumioBoss'
Traceback (most recent call last):
  File "loader.py", line 3, in <module>
  File "runpy.py", line 180, in run_module
  File "runpy.py", line 72, in _run_code
  File "/data/plugins/music_service/volspotconnect/spotify-connect-web/main.py", line 14, in <module>
    from connect import Connect
  File "/data/plugins/music_service/volspotconnect/spotify-connect-web/connect.py", line 9, in <module>
    from console_callbacks import audio_arg_parser, mixer, error_callback, connection_callbacks, debug_callbacks, playback_callbacks, playback_setup
  File "/data/plugins/music_service/volspotconnect/spotify-connect-web/console_callbacks.py", line 24, in <module>
    audio_arg_parser.add_argument('--mixer', '-m', help='alsa mixer name for volume control', default=alsa.mixers()[0])
alsaaudio.ALSAAudioError: Invalid argument [default]
Failed to execute script loader
```
I'm not sure where to start, because the startconnect looks ok:
`--name 'VolumioBoss' --playback_device plughw:1 --mixer 'Digital'  --mixer_device_index 1 --dbrange 45`

There is a mixer and an index defined.

**Comment by Saiyato** [comment]
The above was tested on my Pi3, so I thought, maybe it's just my Pi3. Yesterday evening I re-imaged my Pi2, no DAC and I get the same error:

```
May 30 08:44:47 volumio volumio[1202]: info: Volspotconnect Daemon Started
May 30 08:44:47 volumio volumio[2103]: Established under name 'Volumio'
May 30 08:44:49 volumio volumio[2103]: Loading Spotify library...
May 30 08:44:49 volumio volumio[2103]: Traceback (most recent call last):
May 30 08:44:49 volumio volumio[2103]: File "loader.py", line 3, in <module>
May 30 08:44:49 volumio volumio[2103]: File "runpy.py", line 180, in run_module
May 30 08:44:49 volumio volumio[2103]: File "runpy.py", line 72, in _run_code
May 30 08:44:49 volumio volumio[2103]: File "/data/plugins/music_service/volspotconnect/spotify-connect-web/main.py", line 14, in <module>
May 30 08:44:49 volumio volumio[2103]: from connect import Connect
May 30 08:44:49 volumio volumio[2103]: File "/data/plugins/music_service/volspotconnect/spotify-connect-web/connect.py", line 9, in <module>
May 30 08:44:49 volumio volumio[2103]: from console_callbacks import audio_arg_parser, mixer, error_callback, connection_callbacks, debug_callbacks, playback_callbacks, playback_setup
May 30 08:44:49 volumio volumio[2103]: File "/data/plugins/music_service/volspotconnect/spotify-connect-web/console_callbacks.py", line 24, in <module>
May 30 08:44:49 volumio volumio[2103]: audio_arg_parser.add_argument('--mixer', '-m', help='alsa mixer name for volume control', default=alsa.mixers()[0])
May 30 08:44:49 volumio volumio[2103]: alsaaudio.ALSAAudioError: Invalid argument [default]
May 30 08:44:49 volumio volumio[2103]: Failed to execute script loader
```

So that got me thinking, maybe it's the new Volumio image. Because I was pretty sure the plugin worked before, on both the Pi2 and the Pi3 + DAC.

I'm using the latest image: 2.175; possibly something was changed that breaks spotify-web-connect.

**Comment by Saiyato** [comment]
Just googling a bit and I found this one: https://github.com/Fornoth/spotify-connect-web/issues/67

Creating an asound.conf with the following contents does fix the problem:

```
pcm.!default {
        type hw
        card 0
}

ctl.!default {
        type hw
        card 0
}
```

Not sure where the problem lies, yes in Volumio, but is there some default registration missing? Or?

**Comment by balbuze** [comment]
Weird. The plugin copy the asound.conf if exists but only for the chroot version (armv6).'have you a reason to not use volspotconnect2?

**Comment by Saiyato** [comment]
Yes, last time I checked the repeat and shuffle didn't work. But I can try again :)

What's also weird is that my Pi1 does work... Will investigate further tonight

**Comment by Saiyato** [comment]
Ok it gets weirder, the Pi1 writes an asound.conf file in the plugin...

```
pcm.softvol {
    type            softvol
    slave {
        pcm         "plughw:5,0"
    }
    control {
        name        "SoftMaster"
        card        5
        device      0
    }
max_dB 0.0
min_dB -50.0
resolution 100
}
```
Will re-check with an earlier image and the Pi2. There's more differences comparing the two spotify-web-connect directories.

**Comment by balbuze** [comment]
Yes rpi1 use a chroot version that needs asound.conf to work properly. I let the same line for all models even if not required for Rpi2

**Comment by Saiyato** [comment]
I'll try to investigate the difference between image 2.114 (my previous test version) and 2.175 (the latest)

**Comment by Saiyato** [comment]
Installation on Volumio 2.114 is flawless, no errors whatsoever. It seems something changed in the ALSA mixer config?

**Comment by balbuze** [comment]
You're probably right. I saw someone reporting issue with softvol. Even if it is not directly relevant to the plugin, softvol write a asound file. But I can't check now...

**Comment by Saiyato** [comment]
I found that the problem came to life between 2.156 and 2.163, not sure which builds are in between. Still investigating. This narrows the commits down to any one between the 3rd and 5th of may this year.

**Comment by Saiyato** [comment]
If I set the pcm.!default and ctl.!default to the desired card it works on Volumio 2.163+

For example:
```
pcm.!default {
    type hw
    card 1
}

ctl.!default {
    type hw           
    card 1
}
```

**Comment by Saiyato** [comment]
I've been informed ALSA has been updated, I presume they were talking about libasound (the ALSA lib/API). So I experimented a bit and guess what, renaming libasound.so.2 to something else in the /data/plugins/music_service/volspotconnect/spotify-connect-web directory fixed the problem.

The library seems to be incompatible with the ALSA update, removing it from the plugin defaults it back to /usr/lib/arm-linux..../libasound.so.2

In short: rename libasound.so.2 in the plugin dir and your problem is gone!
